### PR TITLE
Developer Client API Interface

### PR DESCRIPTION
**Pull request by aomi**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  // TODO: we probably don't want to return the Kuali data as-is.
+  const courseDetails = await got(COURSE_DETAIL_URL + pid).json<KualiCourseItem>();
+  // TODO: strip HTML tags from courseDetails.description
+  return courseDetails;
+};
+
+// TODO: change name of this
+export const Demo = async () => {
+  // upon initialization, we download the main Kuali courses JSON file.
+  const kuali = await got(COURSES_URL).json<KualiCourseCatalog[]>();
```
Comment (isaaccormack) [inline src/index.ts:61]: would be helpful for readability if you typed the variables which were returned so the review doesn't have to go to the method to see what is returned

**Comment by isaaccormack** [comment]
It might be easiest for review if you just create a rough doc which compares some of your proposed API interface designs. Its quicker to understand / change and a doc like that will need to be created for end-users eventually.

**Comment by keithradford** [comment]
Let me know what you guys think of the changes in lines 59-75 of index.ts. If it's good we can continue adding the stuff from the documentation we prepared.

File: README.md [hunk 2]
```diff
@@ -9,6 +9,31 @@ Scheduler Scraper is a Node.js library that parses information from [University
 
 As a developer, you would use this to parse HTML and JSON from Kuali and BAN1P which would be retrieved by any method like fetch etc.
 
+# Install
+
+```
+npm install @vikelabs/scheduler-scraper
```
Comment (keithradford) [inline README.md:15]: README should be updated to reflect the current implementations

**Review by keithradford — APPROVED** [review]

**Pull request merged**